### ISSUE TITLE
Adding parameter with trafo a bit tricky

### ISSUE THREAD

**Issue by mb706** [mlr-org/paradox#248]
Suppose a `ParamSet` contains parameters that have a trafo, and one wants to add a new parameter that also has a trafo. Old `ParamHelpers` could do this incrementally, by doing
```r
par.set = c(par.set,
  makeParamSet(
    makeNumericParam("scale_pos_weight", lower = -10, upper = 10,
      trafo = function(x) 2^x)))
```
with paradox this is harder because the `$trafo` is overwritten, so one must write the `$trafo` function completely with all transformations.

**Comment by jakob-r** [comment]
This is totally right. But how often does it occur? Can we prevent the problem of dynamically changing a ParamSet in another way?

In ParamHelpers we spent much time implementing stuff to allow these dynamic changes and realized in the end that we actually don't use it that much. Often it is much easier to do thin in another place.

Btw, I had a paradox ParamSet combine function that did exactly that but it got scraped:
https://github.com/mlr-org/paradox/blob/c7c39fdd7d6efb588b6096e94071420d95618055/R/ParamSet.R#L248-L278

**Comment by berndbischl** [comment]
Isn't this nearly the same issue?
https://github.com/mlr-org/paradox/issues/245

**Comment by berndbischl** [comment]
You can basically achieve exactly what you want Martin trough clean functional composition?
